### PR TITLE
Add relative jacobian

### PR DESCRIPTION
**Pull request by HosameldinMohamed**
Adds a block that computes the `Relative Jacobian`, implementing https://github.com/robotology/idyntree/blob/f9051752ca5a6ca8f3a4f101e4a536fe1eb62c5f/src/high-level/src/KinDynComputations.cpp#L1661-L1663

See #225

### CODE REVIEW

File: toolbox/library/src/RelativeJacobian.cpp [hunk 1]
```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (C) 2018 Istituto Italiano di Tecnologia (IIT)
```
Comment (traversaro) [inline toolbox/library/src/RelativeJacobian.cpp:2]:
```suggestion
 * Copyright (C) 2021 Istituto Italiano di Tecnologia (IIT)
```

**Comment by traversaro** [comment]
Please ignore codacy, it is a false positive as the analyzer don't get Eigen::Map semantics.

File: toolbox/library/include/WBToolbox/Block/RelativeJacobian.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * @brief The wbt::RelativeJacobian class
+ *
+ * @section Parameters
+ *
+ * In addition to @ref wbblock_parameters, wbt::RelativeJacobian requires:
+ *
+ * | Type | Index | Rows  | Cols  | Name  |
+ * | ---- | :---: | :---: | :---: | ----- |
+ * | ::STRING | 0 + WBBlock::NumberOfParameters | 1 | 1 | "Frame" |
```
Comment (traversaro) [inline toolbox/library/include/WBToolbox/Block/RelativeJacobian.h:38]: This does not seem to be consistent with https://github.com/robotology/wb-toolbox/pull/226/files#diff-2b9b96a62b999213b36e92d8d6259e05232da379e1c2a6e722f67ee1f35f8b5cR81 .

File: matlab/library/WBToolboxLibrary_repository.mdl [hunk 3]
```diff
@@ -3184,6 +3206,165 @@ Library {
...
+	      ContentPreviewEnabled   on
+	      Object {
+		$PropName		"MaskObject"
+		$ObjectID		124
+		$ClassName		"Simulink.Mask"
+		Type			"Relative Jacobian"
+		Description		"This block retrieves the Relative Jacobian between two specified frames.\n\nAssuming DoFs is the numbe"
+		"r of internal degrees of freedom of the robot:\n\nInput:\n  - Joint configuration: Vector of size DoFs, representing"
+		" the configuration of the joints.\n\nOutput:\n  - Jacobian: a 6xdofs matrix representing the Jacobian between the sp"
+		"ecified frames written in Mixed representation: origin as child, orientation as parent.\n\nParameters:\n  - Frame na"
```
Comment (traversaro) [inline matlab/library/WBToolboxLibrary_repository.mdl:3226]: I may be missing something, but it is not clear from here wheter "Frame 1" is parent or "Frame 2" is child. Furthermore, I am not sure if parent/child are the best choice of words, as this frames can be also relative to links that are not one the parent of another. Perhaps referenceFrame/frame or reference/leaf could be better names?

**Review by traversaro — CHANGES_REQUESTED** [review]
Some comments on the docs, other then that it seems ok.

File: matlab/library/WBToolboxLibrary_repository.mdl (hunk 3 above)
Comment (HosameldinMohamed) [reply matlab/library/WBToolboxLibrary_repository.mdl:3226]: Yes!

image

Comment (HosameldinMohamed) [reply matlab/library/WBToolboxLibrary_repository.mdl]: Updated in https://github.com/robotology/wb-toolbox/pull/226/commits/c70ea7d4a826fc7d2fdd37aaf7a1634cf7033ec2

File: toolbox/library/src/RelativeJacobian.cpp (hunk 1 above)
Comment (HosameldinMohamed) [reply toolbox/library/src/RelativeJacobian.cpp:2]: Updated in https://github.com/robotology/wb-toolbox/pull/226/commits/c70ea7d4a826fc7d2fdd37aaf7a1634cf7033ec2

File: toolbox/library/include/WBToolbox/Block/RelativeJacobian.h (hunk 2 above)
Comment (HosameldinMohamed) [reply toolbox/library/include/WBToolbox/Block/RelativeJacobian.h]: Updated in https://github.com/robotology/wb-toolbox/pull/226/commits/c70ea7d4a826fc7d2fdd37aaf7a1634cf7033ec2

**Comment by HosameldinMohamed** [comment]
@traversaro I addressed your previous comments, ready for review again.

**Comment by traversaro** [comment]
Great! Can you rebase this branch on master, and update the CHANGELOG ?

**Comment by HosameldinMohamed** [comment]
> Great! Can you rebase this branch on master, and update the CHANGELOG ?

Done in https://github.com/robotology/wb-toolbox/commit/d19eba84f8518e1f00c30296c1cc0747633d6e14

**Review by traversaro — APPROVED** [review]

**Comment by traversaro** [comment]
@HosameldinMohamed do you have any planned future improvements? If not, I think I can proceed with a release before we forget about it.

**Comment by HosameldinMohamed** [comment]
> @HosameldinMohamed do you have any planned future improvements? If not, I think I can proceed with a release before we forget about it.

@traversaro nothing at the moment.

**Comment by HosameldinMohamed** [comment]
Thanks!

**Pull request merged**